spec/GraphQLOverHTTP.md

```diff
@@ -237,120 +311,304 @@ omitted.
 
 GET requests MUST NOT be used for executing mutation operations. If the values
 of {query} and {operationName} indicate that a mutation operation is to be
-executed, the server SHOULD immediately respond with error status code `405`
-(Method Not Allowed) and halt execution. This restriction is necessary to
-conform with the long-established semantics of safe methods within HTTP.
+executed, the server MUST respond with error status code `405` (Method Not
+Allowed) and halt execution. This restriction is necessary to conform with the
+long-established semantics of safe methods within HTTP.
 
 ## POST
 
 A GraphQL POST request instructs the server to perform a query or mutation
 operation. A GraphQL POST request MUST have a body which contains values of the
-request parameters encoded according to the value of `Content-Type` header of
-the request.
+request parameters encoded in one of the officially recognized GraphQL media
+types, or another media type supported by the server.
+
+A client MUST indicate the media type of a request body using the `Content-Type`
+header as specified in [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+A server MUST support POST requests encoded with the `application/json` media
+type (as indicated by the `Content-Type` header).
+
+If the client does not supply a `Content-Type` header with a POST request, the
+server SHOULD reject the request using the appropriate `4xx` status code.
+
+A server MAY support POST requests encoded with and/or accepting other media
+types.
```

Comment:
A non-normative note suggesting that servers which support POST requests with content-type `application/x-www-form-urlencoded`, `multipart/form-data`, or `text/plain` need to take extra care to avoid CSRF may be useful.